### PR TITLE
rasheed - completed core - started extension

### PR DESCRIPTION
**Pull request by desokan**

### CODE REVIEW

File: src/controllers/customer.js [hunk 1]
```diff
@@ -1,53 +1,62 @@
...
+
+    res.status(500).json({ error: e.message });
+  }
+};
+
+const updateCustomer = async (req, res) => {
+  const id = Number(req.params.id);
+  const { name } = req.body;
+  const customer_update = await updateCustomerData(id, name);
+  console.log("xxxxxxxxxxxx", customer_update);
```
Comment (procrastian) [inline src/controllers/customer.js:55]: Nice way to find things easily in the console!

File: src/models/screen.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    },
+  });
+
+  // const movie_with_screen = getMovieDataById(new_screen.id);
+  return new_screen;
+};
+
+module.exports = {
+  createScreenData,
+};
```
Comment (procrastian) [inline src/models/screen.js:17]: nice use of single responsibility thinking to extract the prisma code out into models!

File: src/controllers/customer.js [hunk 3]
```diff
@@ -1,53 +1,62 @@
...
-            error: "Missing fields in request body"
-        })
-    }
+        contact: {
+          create: {
+            phone,
+            email,
+          },
+        },
+      },
```
Comment (procrastian) [inline src/controllers/customer.js:29]: Could this also be extracted into your models file for consistency?

File: src/controllers/movieController.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
+const { Prisma } = require("@prisma/client");
```
Comment (procrastian) [inline src/controllers/movieController.js:1]: naming this and the screen controller files differently to the rest is a little jarring, you could maintain a consistent naming convention across the app so it is easier to follow

File: src/models/customer.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        id: true,
+        customerId: true,
+        phone: true,
+        email: true,
+        createdAt: true,
+        updatedAt: true,
+      },
+    },
+  },
+};
```
Comment (procrastian) [inline src/models/customer.js:20]: could you explain how setting out the contact info like this works? Is it defining each parameter as existing? I'm unsure as to the use of this?

File: src/controllers/movieController.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+//   const movie = await prisma.movie.findMany();
+//   return movie
+// };
+
+const getMovies = async (req, res) => {
+  const { runtimeLt, runtimeGt } = req.query;
+  const movie = await getMoviesData(runtimeLt, runtimeGt);
+  res.json(movie);
+};
+
```
Comment (gid-ctrl) [inline src/controllers/movieController.js:26]: Im assuming this is for the extension?

File: src/controllers/movieController.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  res.status(201).json(new_movie);
+};
+
+const updateMovie = async (req, res) => {
+  const id = Number(req.params.id);
+  const { title, runtimeMins } = req.body;
+  const movie_update = await updateMovieData(id, title, runtimeMins);
+  res.status(201).json({ movie: movie_update });
+};
+
```
Comment (gid-ctrl) [inline src/controllers/movieController.js:45]: really organised code, very easy to read and understand. Great work

File: src/models/movie.js [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+  } else {
+    const movies = await prisma.movie.findMany({
+      include: {
+        screenings: true,
+      },
+    });
+    console.log("neither");
+    return { movies };
+  }
+};
```
Comment (procrastian) [inline src/models/movie.js:67]: Nice logic for handling combinations of run times being provided! Could using a switch statement be useful here for handling the different parameters you want passing in depending on the timings provided?

**Review by procrastian — COMMENTED** [review]
Nice work as always Rasheed! Nice clean code and easy to follow - I can see your thought process and nice to see how you've adapted the codebase for single responsibility using domains/models

File: src/models/screen.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+const createScreenData = async (number) => {
+  const new_screen = await prisma.screen.create({
+    data: {
+      number,
+    },
+  });
+
+  // const movie_with_screen = getMovieDataById(new_screen.id);
+  return new_screen;
+};
```
Comment (gid-ctrl) [inline src/models/screen.js:13]: you can remove this comment

File: src/models/movie.js [hunk 10]
```diff
@@ -0,0 +1,114 @@
+// const { Prisma } = require("@prisma/client");
+const prisma = require("../utils/prisma");
+
+// const getMoviesData = async () => {
+//   const movies = await prisma.movie.findMany();
+//   return {movies}
+// };
```
Comment (gid-ctrl) [inline src/models/movie.js:7]: the comment is not needed

**Pull request closed without merging**